exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/SqlPatternContainsMatcher.java

```diff
@@ -19,44 +19,283 @@
+          }
+        }
+      }
+      return 0;
+    }
+  }
+
+  /** Handles patterns with arbitrary length */
+  private final class MatcherN extends MatcherFcn {
+
+    private MatcherN() {
+      super();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected final int match(int start, int end, DrillBuf drillBuf) {
+
+      if (patternLength == 0) {
+        return 1;
+      }
+
+      final int lengthToProcess = end - start - patternLength + 1;
+      int patternIndex          = 0;
+      final byte firstPattByte  = patternArray[0];
+
+      // simplePattern string has meta characters i.e % and _ and escape characters removed.
+      // so, we can just directly compare.
+      for (int idx = 0; idx < lengthToProcess; idx++) {
+        final byte inputByte = drillBuf.getByte(start + idx);
+
+        if (firstPattByte == inputByte) {
+          for (patternIndex = 1; patternIndex < patternLength; ++patternIndex) {
+            final byte currInByte   = drillBuf.getByte(start + idx + patternIndex);
+            final byte currPattByte = patternArray[patternIndex];
+
+            if (currInByte != currPattByte) {
+              break;
+            }
+          }
+
+          if (patternIndex == patternLength) {
+            break;
+          }
+        }
+      }
+      return patternIndex == patternLength ? 1 : 0;
+    }
+  }
+
+  /**
+   * Boyer-Moore matcher algorithm; excellent for large patterns and for prefix patterns which appear
+   * frequently in the input.
+   */
+  private final class BoyerMooreMatcher extends MatcherFcn {
+    private final int[] offsetTable;
+    private final int[] characterTable;
+
+    private BoyerMooreMatcher() {
+      super();
+
+      this.offsetTable    = makeOffsetTable();
+      this.characterTable = makeCharTable();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected int match(int start, int end, DrillBuf drillBuf)  {
+      if (patternLength == 0) {
+        return 1;
+      }
+
+      final int inputLength = end - start;
+
+      for (int idx1 = patternLength - 1, idx2; idx1 < inputLength;) {
+        for (idx2 = patternLength - 1; patternArray[idx2] == drillBuf.getByte(start + idx1); --idx1, --idx2) {
+          if (idx2 == 0) {
+            return 1;
+          }
+        }
+        // i += pattern.length - j; // For naive method
+        idx1 += Math.max(offsetTable[patternLength - 1 - idx2], characterTable[drillBuf.getByte(start + idx1) & 0xFF]);
+      }
+      return 0;
+    }
+
+    /** Build the jump table based on the mismatched character information **/
+    private int[] makeCharTable() {
+      final int TABLE_SIZE = 256; // This implementation is based on byte comparison
+      int[] resultTable    = new int[TABLE_SIZE];
+
+      for (int idx = 0; idx < resultTable.length; ++idx) {
+        resultTable[idx] = patternLength;
+      }
+
+      for (int idx = 0; idx < patternLength - 1; ++idx) {
+        final int patternValue    = ((int) patternArray[idx]) & 0xFF;
```

Comment:
The cast may be unnecessary. `0xFF` is an `int` which will case the `byte` `patternArray[idx]` to be promoted to `int` implicitly.